File: day-6/part-2.lua
io = require("io")

local input = io.open("input.txt", "r"):read()
local ageCounts = { 0, 0, 0, 0, 0, 0, 0, 0, 0 }

for match in input:gmatch("[^,]+") do
	local index = tonumber(match) + 1
	ageCounts[index] = ageCounts[index] + 1
end

for i = 1, 256 do
	local dead = ageCounts[1]
	
	for j = 2, #ageCounts do
		ageCounts[j - 1] = ageCounts[j]
	end
	
	ageCounts[6 + 1] = ageCounts[6 + 1] + dead
	ageCounts[8 + 1] = dead
end

local sum = 0
for i = 1, #ageCounts do
	sum = sum + ageCounts[i]
end

print(sum)

File: day-6/part-2.c
#include <stdio.h>

int main() {
	size_t t, i, j, ages[9] = { 0 }, sum = 0;
	FILE* input = fopen("input.txt", "r");
	
	do {
		fscanf(input, "%zd", &t);
		
		ages[t]++;
	} while (fgetc(input) == ',');
	
	fclose(input);
	
	for (i = 0; i < 256; i++) {
		ages[(i + 7) % 9] += ages[i % 9];
	}
	
	for (i = 0; i < 9; i++) {
		sum += ages[i];
	}
	
	printf("%zd\n", sum);
	
	return 0;
}
